instrumentation/nocode/src/main/java/com/splunk/opentelemetry/instrumentation/nocode/NocodeMethodInvocation.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.instrumentation.nocode;
+
+import com.splunk.opentelemetry.javaagent.bootstrap.nocode.NocodeRules;
+import io.opentelemetry.instrumentation.api.incubator.semconv.util.ClassAndMethod;
+
+public class NocodeMethodInvocation {
+  private final NocodeRules.Rule rule;
+  private final ClassAndMethod classAndMethod;
+  private final Object thiz;
+  private final Object[] parameters;
+
+  public NocodeMethodInvocation(
+      NocodeRules.Rule rule, ClassAndMethod cm, Object thiz, Object[] parameters) {
+    this.rule = rule;
+    this.classAndMethod = cm;
+    this.thiz = thiz;
+    this.parameters = parameters;
+  }
+
+  public NocodeRules.Rule getRule() {
+    return rule;
+  }
+
+  public Object getThiz() {
+    return thiz;
+  }
+
+  public Object[] getParameters() {
+    return parameters;
+  }
```

Comment:
I don't know how much we care, but I wanted to at least point out that this leaks a mutable array that a caller _could_ change out from under us....making this class not truly immutable.